### ISSUE TITLE
Licence missing

### ISSUE THREAD

**Issue by mrh1997** [B-Con/crypto-algorithms#14]
We would like to integrate the files but need a known licence to know if we are allowed or not.
Could you please add a licence file?

**Comment by jachavez06** [comment]
From readme:

> This code is released into the public domain free of any restrictions. The author requests acknowledgement if the code is used, but does not require it. This code is provided free of any liability and without any quality claims by the author.
> Note that these are not cryptographically secure implementations. They have no resistence to side-channel attacks and should not be used in contexts that need cryptographically secure implementations.

**Comment by Jedarc** [comment]
#12

**Comment by mrh1997** [comment]
My fault. I simply looked for a LICENSE file and didn't check the README, as I expect the README to contain intructions how to use the software.

But as it seems that I am not the only one, maybe adding a LICENCE file would still make sense?
I.e. creative commons (CC0)?

**Comment by Jedarc** [comment]
I agree with you, but unfortunately I haven't seen the author here lately. @B-Con

**Comment by SOesterreicher** [comment]
Hello B-Con,
Public Domain is in some countries problematic. Is there a possibility to assign a permissive license (e.g., MIT, BSD 3-clause) for Commercial use of the code. 
A statement like the following would help:
"For Commercial use this code can be used under the MIT license."

Or do a re-licensing to a permissive license for the whole code. Than a license file would be nice.

Thanks in advance.
Siegfried Oesterreicher

**Comment by Flameeyes** [comment]
I'd like to add to the request for an explicit license — Public Domain is not really well recognized.

To keep with the spirit of your licensing, would you consider using [Unlicense](https://unlicense.org/) ? It's recognized by many organization, although it's not (yet?) OSI-approved.

For others to use this code, it would also be great to have SDPX headers as recommended by REUSE.software — that way there's a clear indication of copyright and license, and makes it easier to import into other projects.
